examples/plot_text.py

```diff
@@ -0,0 +1,157 @@
+DiCoDiLe algorithm.
+
+"""  # noqa
+import matplotlib.pyplot as plt
+import numpy as np
+
+from dicodile import dicodile
+from dicodile.data.images import fetch_letters_pami
+from dicodile.update_d.update_d import tukey_window
+from dicodile.utils.csc import reconstruct
+from dicodile.utils.dictionary import init_dictionary
+from dicodile.utils.viz import display_dictionaries
+
+
+###############################################################################
+# We will first load PAMI image generated from a text of **5000**
+# characters drawn uniformly from the **4** letters **P** **A** **M**
+# **I** and 2 whitespaces and assign it to `X`.
+#
+# We will also load the images of the four characters used to generate
+# `X` and assign it to variable `D`.
+
+X_original, D = fetch_letters_pami()
+
+
+###############################################################################
+# We will work on the copy `X` of the original image and we need to
+# reshape image data `X` and `D` to fit to expected signal shape of
+# `dicodile`:
+#
+# (n_channels, *sig_support)
+
+X = X_original.copy()
+
+X = X.reshape(1, *X.shape)
+D = D.reshape(4, 1, *D.shape[-2:])
+
+# pad `D`
+D = np.pad(D, [(0, 0), (0, 0), (4, 4), (4, 4)])
+
+
+###############################################################################
+# Let's display an extract of the original text image `X_original` and
+# all the images of characters from `D`.
+
+zoom_x = X_original[190:490, 250:750]
+plt.axis('off')
+plt.imshow(zoom_x, cmap='gray')
+
+display_dictionaries(D)
+
+###############################################################################
+# We add a Gaussian white noise with standard deviation σ std (X) and σ = 3
+# to `X`.
+
+std = 3
+rng = np.random.default_rng(None)
+
+X += std * X.std() * rng.standard_normal(X.shape)
+
+###############################################################################
+# We will create a random dictionary of **K = 10** patches from the
+# noisy image.
+
+# set number of patches
+n_atoms = 10
+# set individual atom (patch) size
+atom_support = np.array(D.shape[-2:])
+
+D_init = init_dictionary(X, n_atoms=n_atoms, atom_support=atom_support,
+                         random_state=60)
+
+# window the dictionary, this helps make sure that the border values are 0
+atom_support = D_init.shape[-2:]
+tw = tukey_window(atom_support)[None, None]
+D_init *= tw
+
+###############################################################################
+# Let's display an extract of noisy `X` and random dictionary `D_init`
+# generated from `X`.
+
+zoom_x = X[0][190:490, 250:750]
+plt.axis('off')
+plt.imshow(zoom_x, cmap='gray')
+
+display_dictionaries(D_init)
+
+###############################################################################
+# Add a small noise to avoid having coefficients that are equal which
+# might complicate distributed optimization.
+
+X_0 = X.copy()
+X_0 += X_0.std() * 1e-8 * rng.standard_normal(X.shape)
```

Comment:
I doubt about the necessity of these lines as well. We already add noise at line 64. This exists in `benchmarks/dicodile_text.py` example. Should it stay?

Reply:
Removed as we agreed last week!